Mathlib/Data/Set/Function.lean

```diff
@@ -562,14 +562,21 @@ theorem MapsTo.mem_iff (h : MapsTo f s t) (hc : MapsTo f sᶜ tᶜ) {x} : f x 
 
 section
 
-variable (t f)
+variable (s t f)
```

Comment:
It's often a good idea to avoid explicit arguments in a variable block, *especially* with hypotheses.
Here, `s` is just a type definition - my intuition is not sure if this is a good idea. I'll let a more experienced reviewer comment.

You certainly could do `variable (s) in` just before your theorem. Whether that's better... well, let's wait for advice, I'm not fully sure any more.

Reply:
I don't think it's a problem here because it's a very small section (and indeed this already applies to the current code so I don't think we should consider it a problem). 

Just to be extra clear for Benoit, the issue that could arise is that this would make `s` explicit in a bunch of lemmas below, so when reviewing we have to manually check that it doesn't.

Reply:
I think this is fine.

Reply:
`variable (s) in` is definitely cleaner, and we want to move to that style (and never update binder explicitness globally anymore).
However, it is not your responsibility to do this, since it was already using the suboptimal style before.